File: server/hotrod/src/main/scala/org/infinispan/server/hotrod/ProtocolFlag.scala
package org.infinispan.server.hotrod

/**
 * @author Galder Zamarreño
 */
object ProtocolFlag extends Enumeration {
   type ProtocolFlag = Enumeration#Value
   val NoFlag = Value
   val ForceReturnPreviousValue = Value(0x01)
   val DefaultLifespan = Value(0x02)
   val DefaultMaxIdle = Value(0x04)
   val SkipCacheLoader = Value(0x08)
   val SkipIndexing = Value(0x10)
}

File: server/hotrod/src/main/java/org/infinispan/server/hotrod/ProtocolFlag.java
package org.infinispan.server.hotrod;

/**
 * @author Galder Zamarreño
 */
public enum ProtocolFlag {
   NoFlag(0),
   ForceReturnPreviousValue(1),
   DefaultLifespan(1 << 1),
   DefaultMaxIdle(1 << 2),
   SkipCacheLoader(1 << 3),
   SkipIndexing(1 << 4);

   private final byte value;

   ProtocolFlag(int value) {
      this.value = (byte) value;
   }

   public byte getValue() {
      return value;
   }
}
